server/tracedb/tracedb.go

```diff
@@ -60,7 +60,15 @@ func Factory(repo model.Repository) func(ds model.DataStore) (TraceDB, error) {
 	return f.New
 }
 
-func (f *traceDBFactory) New(ds model.DataStore) (tdb TraceDB, err error) {
+func (f *traceDBFactory) getTraceDBInstance(ds model.DataStore) (TraceDB, error) {
+	var tdb TraceDB
+	var err error
+
+	if ds.IsOTLPBasedProvider() {
+		tdb, err = newCollectorDB(f.repo)
```

Comment:
Where exactly was the bug?

Reply:
It was on our `switch/case` statement. Today we do this:
```go
switch ds.Type {
//...
    case model.DataStoreTypeNewRelic:
    case model.DataStoreTypeLighStep:
    case model.DataStoreTypeDataDog:
    case model.DataStoreTypeOTLP:
        tdb, err = newCollectorDB(f.repo)
//...
}
```

Instead of this:
```go
switch ds.Type {
//...
    case model.DataStoreTypeNewRelic,
         model.DataStoreTypeLighStep,
         model.DataStoreTypeDataDog,
         model.DataStoreTypeOTLP:
        tdb, err = newCollectorDB(f.repo)
//...
}
```

Both are valid statements in golang, but the first one only executes `newCollectorDB` for the type `model.DataStoreTypeOTLP` and assumes that for the other cases we don't have statements.

On the second, as we are aggregating four types on the same `case`, they all execute `newCollectorDB`.

Since we now have the function `IsOTLPBasedProvider` to encapsulate the logic of which providers are OTLP-based, I preferred to create this if and reduce one place to update when we need to add other providers.